# PUI 2017 HW 3-3
## Yuwen Chang (ywc249)

## Question
Is the new bus route imprving commute?

## Null Hypothesis
The commute time now is the same as or longer than before.

$H_0$ : $T_{now} >= T_{before}$

## Alternative Hypothesis
The commute time is shorter than before, with statistical significance.

$H_1$ : $T_{now} < T_{before}$

## Significance level
$\alpha=0.05$

## Z statistic
$Z = \frac{\mu_{pop} - \mu_{sample}}{\sigma / \sqrt{N}}$

In [1]:
import pandas as pd
import numpy as np

In [2]:
# Fetch data
url = "https://raw.githubusercontent.com/fedhere/PUI2017_fb55/master/Lab3_fb55/times.txt"
time = pd.read_csv(url, header=None).values.squeeze()

In [3]:
# Set vars
mu = 36
s = 6
m = time.mean()

In [4]:
# Calculate z score
z = (m - mu) / (s / np.sqrt(len(time)))
z

-2.5563971861666701

In [5]:
# Examine the significance level of the difference
print("z statistic is %.2f"%z)
if abs(z) >= 1.96:
    print("Reject Null Hypothesis")
else:
    print("Unable to reject Null Hypothesis")

z statistic is -2.56
Reject Null Hypothesis


## Conclusion
The Z-value exceeds the threshold (1.96) of single-tail Z-test at the significance level of $\alpha = 0.05$.
We are thus able to reject the null hypothesis because the probability of getting such difference (2.6 mins) is too small (< 5%) based on the Z distribution, given the population mean and standard deviation.